app/oauth2/access-token-request.js

```
@@ -28,12 +28,16 @@ function accessTokenRequest(request) {
     grant_type: 'authorization_code'
   };
   return fetch(config.get('idam.oauth2.token_endpoint') + url.format({ query: params }), options)
-    .then(response => response.status === 200 ? response : response.text().then(text => Promise.reject(new Error(text))))
-    .then(response => response.json())
+    .then(response => {
+
+      if (response.status !== 200) {
+        logger.error('Failed to obtain access token due to an expected response:', response);
```

in case of status != 200 shouldn't we either throw an Error?